### DISCUSSION TITLE
Why was Translation class marked as final?

### DISCUSSION THREAD

**Discussion by NiklasBr** [pimcore/pimcore#10333] [General]
https://github.com/pimcore/pimcore/blob/34dc8b698f8c85175134438945e634ad61a927b9/models/Translation.php#L29

I can't find any comments or documentation why `Pimcore\Model\Translation` was marked `final`, does anyone know why?

**Comment by brusch** [comment]
As far as I can remember the reason was that it's anyway currently not possible to override it, like you can do with models implementing `ElementInterface`. 
What would be your actual use-case?

**Reply by NiklasBr** [reply]
We have been extending that class to allow translating both Names and Values of Classification store Group and Key Definition labels because the Admin translations does not work due to both permissions and UI limitations. It has allowed us to add a new Route with our own controller::action and a new section of the Pimcore admin UI for clients to edit these translations.
